testng-core/src/test/java/org/testng/JarFileUtilsTest.java

```diff
@@ -82,6 +82,27 @@ public void testWithInvalidXmlFile() throws MalformedURLException {
         "Jar suite");
   }
 
+  @Test
+  public void ensureThatSuitesInSutesWorksWithTestnames() throws MalformedURLException {
+    TestNG testNg = new TestNG(false);
+    JarFileUtils utils =
+        newJarFileUtils(
+            Arrays.asList("testng-tests-child2", "testng-tests-child4", "testng-tests-child5"));
+    String[] expectedTestNames =
+        new String[] {"testng-tests-child2", "testng-tests-child4", "testng-tests-child5"};
+    String[] expectedClassNames =
+        new String[] {
+          "org.testng.jarfileutils.org.testng.SampleTest2",
+          "org.testng.jarfileutils.org.testng.SampleTest4",
+          "org.testng.jarfileutils.org.testng.SampleTest5"
+        };
+
+    testNg.setTestNames(Arrays.asList(expectedTestNames));
+    testNg.setXmlPathInJar(jar.getAbsolutePath());
+    testNg.setTestJar(jar.getAbsolutePath());
+    testNg.initializeSuitesAndJarFile();
```

Comment:
@martinaldrin - I dont see any assertions. Was that intentional ?

Reply:
> 

Yes, The purpose of the test is to ensure that a exception is not thrown. I can not add any assert methods.
We just ensure that the method do not check suites in jarfiles here, because that is handled later in JarFileUtiles.
And JarFileUtiles already have unit tests to ensure that this is working as expected, the problem started to aggregate suites in the jarfile here without consider sub suites.

A very similar method exist in class: JarFileUtil method: testngXmlExistsInJar that take care of that part.